### ISSUE TITLE
testing: integrate binary verification into TeamCity

### ISSUE THREAD

**Issue by tbg** [cockroachdb/cockroach#19238]
The release process has a step which [verifies the binaries](https://github.com/cockroachdb/cockroach/wiki/Release-process#check-binaries) but there's currently an option between a fairly shallow manual check or the more extensive [bincheck](https://github.com/benesch/bincheck#testing-a-new-release) courtesy of @benesch.

In light of https://github.com/cockroachdb/cockroach/pull/19225 I wanted to add a check that makes sure that the cluster version matches the binary version, but this isn't going to be effective unless we make bincheck our primary method of verification.

Is that a reasonable course of action?

**Comment by richardwu** [comment]
+1

**Comment by lego** [comment]
There's a lot of value in having CI run these checks. To name a few, preventing mistakes, catching any regressions of compatibility with other platforms, and for extending the verification process to include more tests. Maybe even logic tests and acceptance tests.

Having these checks also be done on CI isn't any worse than already relying on CI to build the binaries.

**Comment by benesch** [comment]
Yes! I've wanted to move bincheck into TeamCity for a while. The blocker has been getting Mac and Windows build agents; previously I was (ab)using Travis and Appveyor for this. 

We could always start with the Linux verification—that's the most important one, anyway—and work our way up from there.

**Comment by lego** [comment]
I notice you've since moved bincheck to the cockroachdb org and removed the manual check notes from the release process. Seems like this should be closed?

I do think we should create an issue to improve bincheck, for example do the cluster version check @tschottdorf  mentioned.

**Comment by benesch** [comment]
I say we leave it open to track making TeamCity run bincheck automatically.

**Comment by rail** [comment]
Reopening this. We still want this to happen.

**Comment by kenliu-crl** [comment]
note for future reference: we want to run this check earlier in the build process than at the end right before a release
